src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,608 @@
+                            source,
+                            invocationExpression,
+                            (variableIdentifier, expression) => SyntaxFactory.ExpressionStatement(
+                                SyntaxFactory.PostfixUnaryExpression(SyntaxKind.PostIncrementExpression, variableIdentifier)), // Generating 'count++;'
+                            SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal(0)),
+                            "count",
+                            out documentUpdate);
+                }
+
+                documentUpdate = default;
+                return false;
+            }
+
+            private bool TryConvertIfInToListInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                out DocumentUpdate documentUpdate)
+            {
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0)
+                {
+                    return TryConvertIfInInvocation(
+                              source,
+                              invocationExpression,
+                              (listIdentifier, expression) => SyntaxFactory.ExpressionStatement(SyntaxFactory.InvocationExpression(
+                                    SyntaxFactory.MemberAccessExpression(
+                                        SyntaxKind.SimpleMemberAccessExpression,
+                                        listIdentifier,
+                                        SyntaxFactory.IdentifierName(nameof(IList.Add))),
+                                    SyntaxFactory.ArgumentList(SyntaxFactory.SingletonSeparatedList(SyntaxFactory.Argument(expression))))),
+                              SyntaxFactory.ObjectCreationExpression(
+                                  methodSymbol.GenerateReturnTypeSyntax().WithAdditionalAnnotations(Simplifier.Annotation),
+                                  SyntaxFactory.ArgumentList(),
+                                  null),
+                               "list",
+                              out documentUpdate);
+                }
+
+                documentUpdate = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                Func<ExpressionSyntax, ExpressionSyntax, StatementSyntax> leafExpressionCreationMethod,
+                ExpressionSyntax initializer,
+                string variableName,
+                out DocumentUpdate documentUpdate)
+            {
+                var parentStatement = invocationExpression.GetAncestorOrThis<StatementSyntax>();
+                if (parentStatement != null &&
+                    !parentStatement.Equals(default) &&
```

Comment:
What is "parentStatement.Equals(default)" for?  I'm confused by it.  You just checked it was not null right above this.